tensorflow/core/ops/image_ops.cc

```diff
@@ -435,6 +435,11 @@ REGISTER_OP("DrawBoundingBoxes")
     .Output("output: T")
     .Attr("T: {float, half} = DT_FLOAT")
     .SetShapeFn([](InferenceContext* c) {
+      ShapeHandle boxes;
+      // boxes: 3-D with shape [batch, num_bounding_boxes, 4]
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(1), 3, &boxes));
+      DimensionHandle unused;
+      TF_RETURN_IF_ERROR(c->WithValue(c->Dim(boxes, 2), 4, &unused));
```

Comment:
I'm not sure this is correct. As I read the code, the rank of images should be 4, and the rank of boxes should be 3.
We can improve it further by checking that the depth dimension is either 1, 3, or 4 or unknown. https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/draw_bounding_box_op.cc#L48